### PR TITLE
feat(monitor): added time in pings list

### LINKED ISSUE #1359: Monitor manual ping debounce / limit (1min)
- store latest manual ping date in monitor
- disable manual ping button using the stored date for 1 min (therefore any other browsers / computers logged in can not abuse)

image

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1359 

### Notes

A summary of what was achieved in this PR

- [ ] Updated ping function to store latest ping information in monitors array
- [ ] disabled the ping button for one minute after click

### CODE REVIEW

File: functions/src/monitor/monitor.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const ping: any = async (projectUid: string, monitorUid: string): Promise
     .collection('projects')
     .doc(projectUid)
     .get();
-  const project: ProjectModel = <ProjectModel> document.data();
+  const project: ProjectModel = <ProjectModel>document.data();
```
Comment (eddiejaoude) [inline functions/src/monitor/monitor.ts]:
```suggestion
  const project: ProjectModel = <ProjectModel> document.data();
```

Comment (eddiejaoude) [reply functions/src/monitor/monitor.ts]: May be ignore 👍

File: functions/src/monitor/monitor.ts [hunk 2]
```diff
@@ -51,6 +51,21 @@ export const ping: any = async (projectUid: string, monitorUid: string): Promise
   };
   pingResult.isValid = !!(pingResult.codeMatched && pingResult.textMatched);
 
+  // Updating monitor with latest ping information
+  monitor['latestPing'] = pingResult;
```
Comment (eddiejaoude) [inline functions/src/monitor/monitor.ts]: Would this be better?

```suggestion
  monitor.latestPing = pingResult;
```

File: web/src/app/monitors/monitors-list/monitors-list.component.ts [hunk 3]
```diff
@@ -52,9 +54,17 @@ export class MonitorsListComponent implements OnInit {
...
+    this.pingProgress = true;
+    this.currentMonitorId = monitorUid;
     this.monitorService
       .pingMonitor(this.projectUid, monitorUid);
+
+    // disable the ping button for 1 second
+    setTimeout(() => {
+      this.pingProgress = false;
+    },
+      60000);
```
Comment (eddiejaoude) [inline web/src/app/monitors/monitors-list/monitors-list.component.ts]: A single like might read better

```typescript
setTimeout(() => this.pingProgress = false, 60000);
```

**Comment by webkhushboo** [comment]
Looks good to me 👍 Please merge the PR

**Pull request merged**